src/NuGet.Clients/VisualStudio.Facade/Events/RestoreEventPublisher.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+
+namespace NuGet.VisualStudio.Facade.Events
+{
+    [Export(typeof(IRestoreEventsPublisher))]
+    [Export(typeof(IRestoreEvents))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class RestoreEventPublisher : IRestoreEventsPublisher, IRestoreEvents
```

Comment:
Actually the implementation itself may be moved to the `SolutionRestoreManager` project. In this case there's no need to declare the facade as a MEF component in VSIX.